PEP 639: Clarify License-File field, slim down terminology & remove future PEPs (#2298)
python/peps: pep-0639.rst

Terminology
===========

This PEP seeks to clearly define the terms it uses, given that some have
multiple established meanings (e.g. import vs. distribution package,
wheel *format* vs. Wheel *project*); are related and often used
interchangeably, but have critical distinctions in meaning
(e.g. :pep:`621` *key* vs. core metadata *field*); are existing concepts
that don't have formal terms/definitions (e.g. project/source metadata vs.
distribution/built metadata, build vs. publishing tools), or are new concepts
introduced here (e.g. license expression/identifier).

> Additionally, the terms from the `PyPA PyPUG Glossary <#pypugglossary_>`_
> (specifically *built/binary distribution*, *distribution package*,
> *project* and *source distribution*), and `SPDX <#spdx_>`_
> (*license identifier*, *license expression*) are included by reference.

Author: Sure, makes sense. I tweaked the above to change the subject to "This PEP", as used in the rest of the PEP outside of the "we"-driven Rejected Ideas section, and move "Project" inside the inline link with canonical capitalization.

```suggestion
This PEP also uses terms defined in the `PyPA PyPUG Glossary <#pypugglossary_>`_
(specifically *built/binary distribution*, *distribution package*,
*project* and *source distribution*), and by the `SPDX Project <#spdx_>`_
(*license identifier*, *license expression*).
```

  and ``LicenseRef-Proprietary`` strings. Examples: ``MIT``, ``GPL-3.0-only``

> **Project** *(Sub: Project Source Tree, Installed Project)*

Reviewer: Minor -- Would it make sense to link the user guide definition here? You expand the definition with source tree vs installed below, which assumes implicit knowledge of the "project" definition. You do explicitly mention you're using project by reference above though, so this is a ease of reading suggestion rather than more critical.

Author: Hmm, I thought about this as I do this here and in Distribution Package as well, but I didn't end up doing it initially. I'll do that now.

Author: BTW, we really need to fix the margins of the definition lists in the Sphinx theme. There is almost no margin/padding between definitions, much less than even normal `<p>`s, making it very hard to read, while there is at least the normal (adequate) spacing if the definition has multiple paragraphs. For now I could hack it with `|`, but I really hate resorting to embedding such style hints in the source markup.

Reviewer: #2300 so this doesn't get lost

Add ``License-File`` field
''''''''''''''''''''''''''

Each instance of the ``License-File`` optional field is specified to contain
the string representation of the path in the project source tree, relative to
the project root directory, of a license-related file.
It is a multi-use field that may appear zero or
more times, each instance listing the path to one such file. Files specified
under this field could include license text, author/attribution information,
or other legal notices that need to be distributed with the package.

As `specified by this PEP <License files in project formats_>`_, its value
is also that file's path relative to the root license directory in both
installed projects and the standardized distribution package types.
In other legacy, non-standard or new distribution package formats and
mechanisms of accessing and storing core metadata, the value MAY correspond
to the license file path relative to a format-defined root license directory,
or MAY be treated as a unique abstract key to access the license file contents

> by another means as specified by the format.

Reviewer: Comment from a non expert -- this paragraph is a little hard to parse. Detailed specifications perhaps matter more, though!

Author: I rewrote it a bit to split into multiple sentences, add a comma and simplify it a touch.

    License-Expression: LicenseRef-Proprietary

Appendix: User Scenarios
========================

The following covers the range of common use cases from a user perspective,
providing straightforward guidance for each. Do note that the following
should **not** be considered legal advice, and readers should consult a

> licensed attorney if they are unsure about the specifics for their situation.

Reviewer: Is "licensed attorney" the best phrasing for world-wide meaning? Perhaps "licenced legal practitioner", although such a turn of phrase will never have the exactness of a specific jurisdiction's nomenclature.

Author: Good point; that's the standard legal disclaimer phraseology in the US but may not be as intelligible worldwide. I'll revise it to your suggestion.
